spot-contracts/contracts/RolloverVault.sol

```diff
@@ -469,7 +469,12 @@ contract RolloverVault is
         // The vault continues to hold the perp dust until the subsequent `swapPerpsForUnderlying` or manual `recover(perp)`.
 
         // Revert if vault liquidity is too low.
-        _enforceUnderlyingBalAfterSwap(underlying_, s.vaultTVL);
+        //  - Absolute balance is strictly greater than `minUnderlyingBal`.
+        //  - Ratio of the balance to the vault's TVL is strictly greater than `minUnderlyingPerc`.
+        uint256 underlyingBal = underlying_.balanceOf(address(this));
```

Comment:
I guess there's a weird corner case where underlying balance could go up (or maybe just remain constant?) after an underlying->perp swap.
e.g. If the vault was sitting on a bunch of As, and the new Zs free those up.
In that case I think we'd want to allow that to go through.

Perhaps the most generalized logic would be:

```
If underlying bal increases or stays the same, continue
else, execute the check logic you have here.
```
That situation would be exceptionally rare, though. Worth it? It would be one extra var and a conditional, I think.

Reply:
cool. So basically if the underlying balances goes down, we then ensure that it doesn't go down too much. If it stays the same or increases its fine..